WeakAuras/GenericTrigger.lua

```diff
@@ -756,6 +761,31 @@ function WeakAuras.ScanEventsInternal(event_list, event, arg1, arg2, ... )
   end
 end
 
+function WeakAuras.ScanEventsInternalToCustom(id, triggernum)
+  Private.StartProfileAura(id);
+  Private.ActivateAuraEnvironment(id);
+  local updateTriggerState = false
+  if trigger_to_custom_events[id] and trigger_to_custom_events[id][triggernum] then
+    for requestingTrigger, bool in pairs(trigger_to_custom_events[id][triggernum]) do
+      if bool then
+        local data = events and events[id] and events[id][requestingTrigger]
+        local updatedTriggerStates = WeakAuras.GetTriggerStateForTrigger(id, triggernum)
+        local allstates = WeakAuras.GetTriggerStateForTrigger(id, requestingTrigger)
+        if data and allstates and updatedTriggerStates then
+          if RunTriggerFunc(allstates, data, id, requestingTrigger, "TRIGGER:"..triggernum, updatedTriggerStates) then
+            updateTriggerState = true
+          end
+        end
+      end
+    end
+  end
```

Comment:
```suggestion
    for requestingTrigger in pairs(trigger_to_custom_events[id][triggernum]) do
      local data = events and events[id] and events[id][requestingTrigger]
      local updatedTriggerStates = WeakAuras.GetTriggerStateForTrigger(id, triggernum)
      local allstates = WeakAuras.GetTriggerStateForTrigger(id, requestingTrigger)
      if data and allstates and updatedTriggerStates then
        if RunTriggerFunc(allstates, data, id, requestingTrigger, "TRIGGER:"..triggernum, updatedTriggerStates) then
          updateTriggerState = true
        end
      end
    end
  end
```

Reply:
looks good

Reply:
I wonder if it wouldn't be better to change trigger function signature to
```
function(allstates, event, triggerNum, statesFromTrigger)
```
aka
```
RunTriggerFunc(allstates, data, id, requestingTrigger, "TRIGGER", triggernum, updatedTriggerStates) 
```

Reply:
That info is carried in the event name. The user would make a trigger like
```LUA
function(event, triggerStates)
  if event == "TRIGGER:3" then
```
Adding the number in also certainly can't hurt, and could save people a step sometimes if they needed the value as a number, but doesn't feel vital. 

Happy to go with whatever people think will feel intuitive though.

Reply:
IMO sending "TRIGGER" & trigger number as an extra argument is consistent with what is done with "CLEU:SPELL_CAST_SUCCESS" or "UNIT_AURA:player", 1 information per argument

Reply:
Oh I see now. Sure, okay.